snapcraft/internal/lxd.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, *, output, source, project_options,
         _verify_remote(remote)
         self._container_name = '{}:snapcraft-{}'.format(remote, container_name)
         # Use the server architecture to avoid emulation overhead
-        kernel = self._get_remote_info()['environment']['kernel_architecture']
+        kernel = self._get_remote_info()['environment']['architectures'][0]
```

Comment:
This is riskier I think.
How do you know that the first architecture is always the kernel architecture? And I'm not sure if yaml.load guarantees the order in lists.
I was thinking of just trying with both values, like:

```
   try:
      kernel = self._get_remote_info()['environment']['kernel_architecture']
   except KeyError:
      kernel = self._get_remote_info()['environment']['kernelarchitecture']
```